near-abi-client-macros/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+use convert_case::{Case, Casing};
+use near_abi_client::__private::{generate_abi_client, read_abi};
+use quote::format_ident;
+use std::path::PathBuf;
+
+#[proc_macro]
+pub fn near_abi_client(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let abi_def = syn::parse_macro_input!(tokens as AbiDef);
+
+    let near_abi = read_abi(PathBuf::from(&abi_def.abi_path.value()));
+
+    let contract_name = abi_def.name.map(|n| format_ident!("{}", n)).or_else(|| {
+        near_abi
+            .metadata
+            .name
+            .clone()
+            .map(|n| format_ident!("Ext{}", n.to_case(Case::UpperCamel)))
```

Comment:
I wonder if we shouldn't re-use the `Ext` semantics for client. It can already be an unintuitive name and also might be confusing since the SDK codegen is the same name?

Reply:
Fair. Changed this to `{}Client`, which should suit better.